File: QuizGame-WebAPI/Models/Question.cs
namespace QuizGame.Models;

public class Question
{
    public int Id {get; set;}
    public string QuestionText {get; set;}
    public string? QuestionImage {get; set;}
    public int SecondsTimeout {get; set;} = 0;
    public double RelativeScore {get; set;} = 1;
    public Answer CorrectAnswer {get; set;}
    public IEnumerable<Answer> IncorrectAnswers {get; set;}
    // public Quiz Quiz {get; set;}

    internal Question()
    {
        
    }
    public Question(string text, CorrectAnswer answer, IEnumerable<IncorrectAnswer> answers)
    {
        QuestionText = text;
        CorrectAnswer = answer;
        IncorrectAnswers = answers;
    }
}

File: QuizGame-Client/src/app/models/question.ts
import { Answer } from "./answer";
import { User } from "./user";

export interface Question{
    id: number;
    questionText: string;
    questionImage: string;
    secondsTimeout: number;
    relativeScore: number;
    category: string;
    createdAt: Date;
    correctAnswer: Answer;
    incorrectAnswers: Answer[];
    owner: User;
    assignedQuizzes: number[];
}